lib/dor/models/releaseable.rb

```diff
@@ -79,7 +79,7 @@ def get_self_release_tags(tags)
     def get_release_tags_for_item_and_all_governing_sets
       return_tags = release_nodes || {}
       collections.each do |collection|
-        return_tags = combine_two_release_tag_hashes(return_tags, Dor::Item.find(collection.id).get_release_tags_for_item_and_all_governing_sets) # recurvise so parents of parents are found
+        return_tags = combine_two_release_tag_hashes(return_tags, Dor::Item.find(collection.id).get_release_tags_for_item_and_all_governing_sets) unless collection.id == id # recurvise so parents of parents are found, but avoid an infinite loop if the collection references itself
```

Comment:
Ideally these release tag hashes would have a thin class of their own, a descendant of Hash overriding `merge`, and then all the semantics here would be pretty clear.